app/src/main/java/app/revanced/manager/ui/viewmodel/SelectedAppInfoViewModel.kt

```diff
@@ -62,7 +62,9 @@ class SelectedAppInfoViewModel(input: Params) : ViewModel(), KoinComponent {
         viewModelScope.launch(Dispatchers.Default) {
             if (!persistConfiguration) return@launch // TODO: save options for patched apps.
 
-            state.value = optionsRepository.getOptions(selectedApp.packageName)
+            withContext(Dispatchers.Main) {
+                state.value = optionsRepository.getOptions(selectedApp.packageName)
+            }
```

Comment:
Thank you for catching this. It would be better to remove the `Dispatchers.Default` parameter from the `launch` call and do something like `state.value = withContext(Dispatchers.Default) { optionsRepository.getOptions() }` instead. This will run the database query on another thread, which was the original goal of the code.

Reply:
```
viewModelScope.launch {
    if (!persistConfiguration) return@launch // TODO: save options for patched apps.

    state.value = withContext(Dispatchers.Default) { optionsRepository.getOptions(selectedApp.packageName) }
}
```
This still causes it to crash. Did I do something wrong?

Reply:
You did everything correctly. It appears accessing `selectedApp.packageName` in the `Default` dispatcher is causing the issue, since that is also a compose state. Strange behavior by compose. Lets just get rid of the `withContext` instead I guess.

Reply:
```
viewModelScope.launch {
    if (!persistConfiguration) return@launch // TODO: save options for patched apps.

    val packageName = selectedApp.packageName
    state.value = withContext(Dispatchers.Default) { optionsRepository.getOptions(packageName) }
}
```

Would this work if we wanted to keep `withContext`?

Reply:
That would work